File: src/main/java/com/smartpark/controller/WalletController.java
package com.smartpark.controller;

import com.smartpark.dto.ApiResponse;
import com.smartpark.entity.User;
import com.smartpark.exception.ResourceNotFoundException;
import com.smartpark.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.*;

import java.math.BigDecimal;
import java.security.Principal;
import java.util.Map;

@RestController
@RequestMapping("/api/wallet")
@RequiredArgsConstructor
public class WalletController {

    private final UserRepository userRepository;

    @GetMapping("/balance")
    @PreAuthorize("hasAnyRole('DRIVER','OWNER','ADMIN')")
    public ResponseEntity<ApiResponse<Map<String, BigDecimal>>> getWalletBalance(Principal principal) {
        User user = userRepository.findByEmail(principal.getName())
                .orElseThrow(() -> new ResourceNotFoundException("User not found"));

        BigDecimal balance = user.getWalletBalance() != null ? user.getWalletBalance() : BigDecimal.ZERO;
        return ResponseEntity.ok(ApiResponse.<Map<String, BigDecimal>>builder()
                .success(true)
                .message("Wallet balance fetched")
                .data(Map.of("walletBalance", balance))
                .build());
    }
}

File: backend/src/controllers/walletController.js
const { prisma } = require('../config/db');

class WalletController {
  // GET /api/wallet/balance
  async getWalletBalance(req, res, next) {
    try {
      const userId = req.user.id;

      const user = await prisma.user.findUnique({
        where: { id: userId },
        select: { id: true, email: true, name: true, walletBalance: true },
      });

      if (!user) {
        return res.status(404).json({ success: false, message: 'User not found' });
      }

      res.json({
        success: true,
        message: 'Wallet balance fetched successfully',
        data: {
          walletBalance: Number(user.walletBalance || 0),
          userId: Number(user.id),
          email: user.email,
          name: user.name,
        },
      });
    } catch (err) {
      next(err);
    }
  }
}

module.exports = new WalletController();
